pkg/authenticator/jwt/requests.go

```diff
@@ -0,0 +1,39 @@
+package jwt
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+)
+
+// AuthenticateRequest sends an authenticate request
+func AuthenticateRequest(authnURL string, account string, username string, jwtToken string) (*http.Request, error) {
+	var authenticateURL string
+
+	if len(username) > 0 {
+		authenticateURL = fmt.Sprintf("%s/%s/%s/authenticate", authnURL, account, url.QueryEscape(username))
+	} else {
+		authenticateURL = fmt.Sprintf("%s/%s/authenticate", authnURL, account)
+	}
+
+	var err error
+	var req *http.Request
+
+	log.Debug(log.CAKC046, authenticateURL)
+
+	formattedJwt := fmt.Sprintf("jwt=%s", jwtToken)
+	requestBody := strings.NewReader(formattedJwt)
+
+	if req, err = http.NewRequest("POST", authenticateURL, requestBody); err != nil {
+		return nil, log.RecordedError(log.CAKC023, err)
+	}
+
+	req.Header.Set("Content-Type", "text/plain")
+	req.Header.Set("Content-Length", string(len(formattedJwt)))
+	req.Header.Set("User-Agent", "k8s")
```

Comment:
Do we want to add support for [accept-encoding](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding) header?
Conjur is able to return access token already encoded base64 instead of plain json ([1](https://github.com/cyberark/conjur/blob/27f2107987f1fceb73d6ebbbc3fb37bcba6aec90/app/controllers/authenticate_controller.rb#L217), [2](https://github.com/cyberark/conjur/blob/27f2107987f1fceb73d6ebbbc3fb37bcba6aec90/app/controllers/authenticate_controller.rb#L336))

Reply:
Currently the components that use authn-k8s-client do the encode themselves . I think this is good idea but not in scope. I wanted to be aligned with the auhtn-k8s flow